cirq-google/cirq_google/devices/grid_device.py

```diff
@@ -161,28 +181,27 @@ def _build_gateset_and_gate_durations(
 def _build_compilation_target_gatesets(
     gateset: cirq.Gateset,
 ) -> Sequence[cirq.CompilationTargetGateset]:
-    """Detects compilation target gatesets based on what gates are inside the gateset.
-
-    If a device contains gates which yield multiple compilation target gatesets, the user can only
-    choose one target gateset to compile to. For example, a device may contain both SYC and
-    SQRT_ISWAP gates which yield two separate target gatesets, but a circuit can only be compiled to
-    either SYC or SQRT_ISWAP for its two-qubit gates, not both.
-
-    TODO(#5050) when cirq-google CompilationTargetGateset subclasses are implemented, mention that
-    gates which are part of the gateset but not the compilation target gateset are untouched when
-    compiled.
-    """
-
-    # TODO(#5050) Subclass core CompilationTargetGatesets in cirq-google.
+    """Detects compilation target gatesets based on what gates are inside the gateset."""
 
+    # Include a particular target gateset if the device's gateset contains all required gates of
+    # the target gateset.
+    # Set all remaining gates in the device's gateset as `additional_gates` so that they are not
+    # decomposed in the transformation process.
     target_gatesets: List[cirq.CompilationTargetGateset] = []
-    if cirq.CZ in gateset:
-        target_gatesets.append(cirq.CZTargetGateset())
-    if ops.SYC in gateset:
+    if all(gate_family in gateset.gates for gate_family in _CZ_TARGET_GATES):
+        target_gatesets.append(
+            transformers.GoogleCZTargetGateset(
+                additional_gates=(list(gateset.gates - set(_CZ_TARGET_GATES)))
```

Comment:
nit: remove extra parens
```suggestion
                additional_gates=list(gateset.gates - set(_CZ_TARGET_GATES))
```